src/libraries/System.Private.CoreLib/src/System/Array.cs

```diff
@@ -1271,6 +1269,8 @@ public static int LastIndexOf(Array array, object? value, int startIndex, int co
                         case CorElementType.ELEMENT_TYPE_R8:
                             result = GenericLastIndexOf<double>(array, value, adjustedIndex, count);
                             break;
+
+                            result = GenericBinarySearch<IntPtr>(array, value, adjustedIndex, count);
```

Comment:
There is some dead code here, likely just an issue in undoing the previous change